### PR TITLE
Registrar: Consolidate repetitive checking of input parameters

### PR DESCRIPTION
**Pull request by stefanberger**
Consolidate the repetitive checking of input parameters into reusable functions.

### CODE REVIEW

File: keylime/registrar_common.py [hunk 1]
```diff
@@ -171,6 +171,36 @@ def log_message(self, format: str, *args: Any) -> None:
 
 
 class UnprotectedHandler(BaseHTTPRequestHandler, SessionManager):
+    def __validate_input(self, method: str) -> Optional[str]:
+        rest_params = web_util.get_restful_params(self.path)
+        if rest_params is None:
+            web_util.echo_json_response(self, 405, "Not Implemented: Use /version/ interface")
```
Comment (stefanberger) [inline keylime/registrar_common.py:177]: needs to be `/agents/` not `/version/`

File: keylime/registrar_common.py [hunk 2]
```diff
@@ -171,6 +171,36 @@ def log_message(self, format: str, *args: Any) -> None:
...
+            return None
+
+        if not web_util.validate_api_version(self, cast(str, rest_params["api_version"]), logger):
+            return None
+
+        if "agents" not in rest_params:
+            web_util.echo_json_response(self, 400, "uri not supported")
+            logger.warning("PUT agent returning 400 response. uri not supported: %s", self.path)
```
Comment (stefanberger) [inline keylime/registrar_common.py:185]: must use `method` here ...

**Review by aplanas — APPROVED** [review]

File: keylime/registrar_common.py [hunk 3]
```diff
@@ -182,6 +171,36 @@ def log_message(self, format: str, *args: Any) -> None:
...
+
+        # If the agent ID is not valid (wrong set of characters), just
+        # do nothing.
+        if not validators.valid_agent_id(agent_id):
+            web_util.echo_json_response(self, 400, "agent_id is not valid")
+            logger.error("%s received an invalid agent ID: %s", method, agent_id)
+            return None
+
+        return agent_id
+
```
[the remark this answers is not in the document]
Comment (stefanberger) [reply keylime/registrar_common.py]: I now added 2 more patches to consolidate __validate_input() into one common BaseClass as you suggested.

**Review by aplanas — APPROVED** [review]

**Comment by maugustosilva** [comment]
Please rebase and we can merge

**Pull request merged**